main.cpp

```diff
@@ -0,0 +1,170 @@
+#include <string.h>
+#include "analysis_tool.h"
+
+
+
+int main(int argc, char* argv[]) {
+
+  string arg1 = "";  // filepath
+  string arg2 = "";  // method provided
+  string arg3 = "";  // additional method provided depending on what arg2 method is called
+
+  // argc will be at least one automatically because arg[0] is equal to executable filepath
+  if(argc > 1){
+    arg1 = static_cast<string>(argv[1]);
+    if(argc > 2) {
+      arg2 = static_cast<string>(argv[2]);
+      if (argc > 3) {
+        arg3 = static_cast<string>(argv[3]);
+      }
+    }
+  }
+
+  arg1 = validateArg1(arg1);
+  arg2 = validateArg2(arg2);
+  //arg3 = validateArg3(arg2, arg3);
+
+  int num_lines = getFileNumLines(arg1);
+  ifstream file(arg1);
+
+  file.seekg(0L, file.beg); // should return the read position to beginning of file
+
+  // Store lines in array of strings
+  string lines[num_lines];
+  for(int i = 0; i < num_lines; i++){
+    getline(file, lines[i]);
+    lines[i] = tokenize(lines[i]);
+  }
+
+  file.close();
+
+  // run tool depending on user selection
+ // if(arg2 == "search"){
+  //  search(file, arg3);
+ // }else if(arg2 == "sort"){
+  //  sort(file, arg3);
+ // }else
+  //if(arg2 == "count"){
+    //count(file);
+  //}
+  //file.close();
+  return 0;
+}
+
+/*
+ * Argument 1 one should hold a valid filepath upon return
+ */
+string validateArg1(string arg){
```

Comment:
I like how you validate each argument. Making sure that the file exists is a good idea.